runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/config/HttpBindingConfig.java

```diff
@@ -205,6 +203,9 @@ private List<HttpRequestType> createRequests(
             {
                 HttpRequestType requestType = new HttpRequestType();
                 requestType.path = request.path;
+                String pattern = String.format("^%s/?(\\?(?<query0>.*))?$",
+                    requestType.path.replaceAll("\\{([a-zA-Z_]+)\\}", "(?<$1>.+?)"));
+                requestType.pathPattern = Pattern.compile(pattern);
```

Comment:
Let's go further and instantiate the `Matcher` instead.

```
requestType.pathMatcher = Pattern.compile(pattern).matcher("");
```

Then later we can use `requestType.pathMatcher.reset(path)` to use the matcher.